### PR TITLE
Reduce Fedora traffic by removing download links from catalog

### LINKED ISSUE #1097: Add DOI links to Solr document
From https://github.com/ualbertalib/HydraNorth/issues/1091#issuecomment-219857973 : search results page rendering is slowed by the need to discover DOI urls for each item. Since they aren't in the Solr index, they have to be retrieved from Fedora. So: add them to metadata.rb, and confirm that this fixes the performance hit.

### PR DESCRIPTION
**Pull request by mbarnett**
also includes small tweaks to generic file page that may improve performance. Tests related
to doi links appearing on results pages are made pending. Closes #1097

### CODE REVIEW

File: app/views/generic_files/_show_actions.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      <%= render_download_link %>
+      <% if Sufia.config.analytics %>
+        &nbsp;|&nbsp;
+        <%= link_to "Analytics", sufia.stats_generic_file_path(@generic_file), id: 'stats' %>
+      <% end %>
+      <% if can? :edit, @generic_file %>
+          &nbsp;|&nbsp;
+          <%= link_to "Edit", sufia.edit_generic_file_path(@generic_file) %>
+      <% end %>
+    </p>
```
Comment (pgwillia) [inline app/views/generic_files/_show_actions.html.erb]: removes 
```
      <% if can?(:create, FeaturedWork) && @generic_file.public? %>
        <% if FeaturedWork.can_create_another? && !@generic_file.featured?%>
          &nbsp;|&nbsp;
          <%= link_to "Feature", sufia.featured_work_path(@generic_file, format: :json), data: {behavior: 'feature'} %>
        <% elsif @generic_file.featured? %>
          &nbsp;|&nbsp;
          <%= link_to "Unfeature", sufia.featured_work_path(@generic_file, format: :json), data: {behavior: 'unfeature-page'} %>
        <% end %>
      <% end %>
```

**Comment by pgwillia** [comment]
Many whitespace changes.  Discovered github's hid whitespace changes ?w=1

File: app/views/generic_files/_show_actions.html.erb (hunk 1 above)
Comment (mbarnett) [inline app/views/generic_files/_show_actions.html.erb]: Yeah, that's intentional in this override, to clarify. We're not using FeaturedWork, and ```public?``` forces extra Fedora traffic, slowing down the page.

**Comment by mbarnett** [comment]
Also re: whitespace yeah Vim users among us may want to look into http://www.vim.org/scripts/script.php?script_id=3967 to help stop committing extraneous whitespace. 

Rubocop will start flagging this once enabled.

**Comment by pgwillia** [comment]
Thought I was being all kinds of clever by using this as set up to visually inspect this change in the web ui
```
rake migration:dataverse_objects['spec/fixtures/migration/test-metadata/dataverse/']
```
But the resulting object doesn't have use the doi as a link from the results or object page.
```
irb(main):001:0> gf = GenericFile.find('8336h188j')
  LocalAuthority Load (0.2ms)  SELECT  `local_authorities`.* FROM `local_authorities` WHERE `local_authorities`.`name` = 'geonames_cities' LIMIT 1
  DomainTerm Load (0.2ms)  SELECT  `domain_terms`.* FROM `domain_terms` WHERE `domain_terms`.`model` = 'generic_files' AND `domain_terms`.`term` = 'spatial' LIMIT 1
  LocalAuthority Exists (0.2ms)  SELECT  DISTINCT 1 AS one FROM `local_authorities` INNER JOIN `domain_terms_local_authorities` ON `local_authorities`.`id` = `domain_terms_local_authorities`.`local_authority_id` WHERE `domain_terms_local_authorities`.`domain_term_id` = 1 AND `local_authorities`.`id` = 1 LIMIT 1
ActiveFedora: loading fedora config from /var/www/sites/hydranorth/config/fedora.yml
ActiveFedora: loading solr config from /var/www/sites/hydranorth/config/solr.yml
Attempted to init base path `dev`, but it already exists
=> #<GenericFile id: "8336h188j", mime_type: nil, format_label: [], file_size: [], last_modified: [], filename: [], original_checksum: [], rights_basis: [], copyright_basis: [], copyright_note: [], well_formed: [], valid: [], status_message: [], file_title: [], file_author: [], page_count: [], file_language: [], word_count: [], character_count: [], paragraph_count: [], line_count: [], table_count: [], graphics_count: [], byte_order: [], compression: [], color_space: [], profile_name: [], profile_version: [], orientation: [], color_map: [], image_producer: [], capture_device: [], scanning_software: [], exif_version: [], gps_timestamp: [], latitude: [], longitude: [], character_set: [], markup_basis: [], markup_language: [], bit_depth: [], channels: [], data_format: [], offset: [], frame_rate: [], label: nil, depositor: "dit.application.test@ualberta.ca", relative_path: nil, import_url: nil, part_of: [], resource_type: ["Dataset"], title: ["Shape File Index to the Sectional Maps, 1917 [of Western Canada, new style, 1905-1955]."], creator: ["Gaffield, Chad", "Baskerville, Peter", "St-Hilaire, Marc", "Amrhein, Carl", "Cadigan, Sean", "Moldofsky, Byron", "Darroch, Gordon", "Bellavance, Claude", "Normand, France"], contributor: [], description: ["This item is a resource in the University of Alberta Libraries' Dataverse Network. Access this item in Dataverse by clicking on the DOI link. | Kind of data: Geospatial Data | This .shp file was derived from a georeferenced scan of the Index to the Sectional Maps, 1917 [of Western Canada, new style, 1905-1955], published by the Department of the Interior February 1, 1917. The original georeferenced scan and world file is included in the study for reference purposes."], rights: "", publisher: ["University of Alberta Libraries"], date_created: "2014", date_modified: Thu, 02 Jun 2016 16:06:10 +0000, subject: ["Prairie Provinces", "Western Canada", "Maps", "Topographic"], language: nil, identifier: ["doi:10.7939/DVN/10161"], related_url: nil, source: nil, proxy_depositor: nil, on_behalf_of: nil, date_uploaded: Thu, 02 Jun 2016 16:06:10 +0000, license: nil, trid: nil, ser: nil, temporal: [], spatial: ["Canada"], is_version_of: nil, unicorn: nil, fedora3uuid: nil, fedora3handle: nil, ingestbatch: "6969z076v", hasCollection: ["Dataverse Datasets"], belongsToCommunity: ["cf95jb44s"], hasCollectionId: ["wm117p010"], year_created: "2014", remote_resource: "dataverse", degree_grantor: nil, dissertant: nil, supervisor: [], committee_member: [], department: [], specialization: nil, date_submitted: nil, date_accepted: nil, graduation_date: nil, alternative_title: [], thesis_name: nil, thesis_level: nil, proquest: nil, abstract: nil, batch_id: "th83kz323", embargo_id: nil, lease_id: nil>
```
Wondering if this is what @pbinkley was talking about for https://github.com/ualbertalib/HydraNorth/pull/1116

**Comment by pbinkley** [comment]
We would like to deploy this on era-test and see what it looks like if the code were deployed before the solr index was updated (i.e. the app wanted to use links pulled out of solr, but those links weren't there yet). Could we live with that for the length of time it would take to reindex the items with DOIs? (i.e. the Dataverse items)? If so, we could save ourselves a deployment.

**Comment by leahvanderjagt** [comment]
So the only links that would not be discovered via the solr index would be
Dataverse items? If so, I think you should try it, but you might want to
run it past Larry/Anna/James. Let me know if I can help facilitate that
conversation.

Best
Leah

On Fri, Jun 3, 2016 at 4:31 PM, Peter Binkley <notifications@github.com>
wrote:

> We would like to deploy this on era-test and see what it looks like if the
> code were deployed before the solr index was updated (i.e. the app wanted
> to use links pulled out of solr, but those links weren't there yet). Could
> we live with that for the length of time it would take to reindex the items
> with DOIs? (i.e. the Dataverse items)? If so, we could save ourselves a
> deployment.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/pull/1157#issuecomment-223710202>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFieXN7VtElJAVthRagKBlc0WsbUenTxks5qIKsngaJpZM4IqDJ9>
> .
>



-- 
___________________________________________________________




*Leah VanderjagtDigital Repository Services Librarian University of
Albertat. 780.492.3851 *
*leahv@ualberta.ca <leahv@ualberta.ca> *

**Comment by wyman10** [comment]
Dataverse can live with that... How long will it be before the re-index?

**Comment by mbarnett** [comment]
> We would like to deploy this on era-test and see what it looks like if the code were deployed before the solr index was updated (i.e. the app wanted to use links pulled out of solr, but those links weren't there yet).

@pbinkley 

There seems to be confusion here. DOI links on item pages will continue to look as before, because until the Solr index is updated, those links are fetched out of Fedora.

That's *why* the links had to be removed from the search results page. If they were left on there, they'd all have continued to be fetched out of Fedora until the Solr index was updated, meaning the page would have continued to be slow.

**Pull request merged**